pkg/storage/replica_proposal.go

```diff
@@ -600,22 +609,6 @@ func (r *Replica) handleLocalEvalResult(
 	// Non-state updates and actions.
 	// ======================
 
-	if originReplica.StoreID == r.store.StoreID() {
```

Comment:
I think maybe a more direct assertion here makes sense. Otherwise we're just going to assert the lResult is empty, and then only if `shouldAssert` is true. Might be better to call it out directly and add a comment.